include/odp/api/spec/ipsec.h

```diff
@@ -1601,6 +1635,22 @@ int odp_ipsec_result(odp_ipsec_packet_result_t *result, odp_packet_t packet);
  */
 int odp_ipsec_status(odp_ipsec_status_t *status, odp_event_t event);
 
+/**
+ * IPSEC DEBUG API to set IPSEC SA parameter type field with new parameter value.
+ *
+ * This operation allows application to modify IPSEC SA parameters not otherwise
+ * exposed to the application. For IPSEC operations, fields like sequence number
+ * is not available to the application in the datapath. The API allows
+ * application to debug and verify various features provided by the platform.
+ *
+ * @param          sa            IPSEC SA to be updated
+ * @param          sa_param      IPSEC SA param to update
+ *
+ * @return 0      On success
+ * @retval <0     On failure
+ */
+int odp_ipsec_debug_sa_update(odp_ipsec_sa_t sa, odp_ipsec_debug_sa_param_t sa_param);
```

Comment:
Any limitations on usage of this call? When this function can be called: when traffic is flowing, only before starting to use the SA, etc?

Capability needed to tell if implementation support IPsec debug functions.

Reply:
Updated the following in V2:
- Clarified the usage of the API with the following statement:
```
+ * It is the responsiblity of the application to make sure that no traffic is   
+ * flowing through the SA when the API is called. Otherwise, this operation may 
+ * lead to undefined behavior.
```
- Added additional odp_ipsec_debug_capability_t in the existing odp_ipsec_capability_t structure.

Reply:
I previously suggested having this debug stuff as an optional part of SA creation, but after the addition of the capabilities and the restriction on when the update function may be called, I think I am ok with the separate update function.

Reply:
As discussed on the call, may be "test" would be better name to use for this functionality than "debug", as user would use the API for better test coverage (and not always for debugging).

Reply:
Ack. Updated the PR to use "test" instead of "debug" in v4.